index.js

```diff
@@ -37,6 +43,69 @@ exports.activate = ({ subscriptions }) => {
 		},
 	);
 
+	client.onReady().then(() => {
+		/**
+		 * Map of registered formatters by language ID.
+		 * @type {Map<string, { dispose(): any }>}
+		 */
+		const registeredFormatters = new Map();
+
+		client.onNotification('stylelint/languageIdsAdded', (/** @type {string[]} */ langIds) => {
```

Comment:
Could you explain why this is implemented in the extension while, unless I'm missing something, the server could just send a [dynamic registration request](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#documentFormattingRegistrationOptions) which would work across all clients?

Reply:
Had no idea that was an option; when I implemented this I was going off of the [blog post](https://code.visualstudio.com/blogs/2016/11/15/formatters-best-practices) that was linked in #25. I took a cursory look and it sounds like you're right, this might do the trick. Potential refactor?

Reply:
Yes, as long as the server uses recent enough libraries, dynamic registration would be preferred as many people are using this server outside of the VSCode. And obviously the idea of language server is that it's editor agnostic.

Reply:
Created an issue to track this refactor: #241. Can't guarantee when it will be addressed, however, since there's a number of [other priorities at the moment](https://github.com/stylelint/vscode-stylelint/issues/233).